examples/internal/proto/examplepb/stream.pb.gw.go

```diff
@@ -306,12 +305,25 @@ func RegisterStreamServiceHandlerClient(ctx context.Context, mux *runtime.ServeM
 			runtime.HTTPError(ctx, mux, outboundMarshaler, w, req, err)
 			return
 		}
-		resp, md, err := request_StreamService_BulkEcho_0(annotatedContext, inboundMarshaler, client, req, pathParams)
+
+		resp, md, reqErrChan, err := request_StreamService_BulkEcho_0(annotatedContext, inboundMarshaler, client, req, pathParams)
 		annotatedContext = runtime.NewServerMetadataContext(annotatedContext, md)
 		if err != nil {
 			runtime.HTTPError(annotatedContext, mux, outboundMarshaler, w, req, err)
 			return
 		}
+		go func() {
+			for {
+				err := <-reqErrChan
+				if err != nil {
+					runtime.HTTPError(annotatedContext, mux, outboundMarshaler, w, req, err)
+					if err := resp.CloseSend(); err != nil {
+						grpclog.Errorf("Failed to terminate client stream: %v", err)
+					}
+					return
+				}
+			}
```

Comment:
This loop should be over the channel itself, so it terminates when the channel is closed
```suggestion
			for err := range reqErrChan {
				if err != nil {
					runtime.HTTPError(annotatedContext, mux, outboundMarshaler, w, req, err)
					if err := resp.CloseSend(); err != nil {
						grpclog.Errorf("Failed to terminate client stream: %v", err)
					}
					return
				}
			}
```

Reply:
Makes sense. But since it seems that there will ever only be one error received before the channel closing, maybe I should just remove the for loop to make that clear?

Reply:
I like that the loop works both if the channel closes without an error and if there is one error. If there is more than one error then this will likely fail to do anything, so I still think the for loop is best. Maybe I'm missing something?